WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -300,12 +300,16 @@ private extension CollectOrderPaymentUseCase {
     ///
     func handleSuccessfulPayment(capturedPaymentData: CardPresentCapturedPaymentData,
                                  onCompletion: @escaping (Result<CardPresentCapturedPaymentData, Error>) -> ()) {
+        let orderDurationRecorder = OrderDurationRecorder.shared
```

Comment:
How temporary is this? If we're going to leave it in the codebase, I think there's value in setting it up for testability...

It's not much, and I know there aren't tests for this UseCase yet (there should be for the legacy one), but providing the singleton as a passed in property which conforms to a protocol seems like a worthwhile change.